src/sentry/sentry_metrics/indexer/cache.py

```diff
@@ -25,7 +26,8 @@ def randomized_ttl(self) -> int:
 
     def make_cache_key(self, key: str, cache_namespace: str) -> str:
         hashed = md5_text(key).hexdigest()
-        return f"indexer:org:str:{cache_namespace}:{hashed}"
+        formatted_partition_key = f":{self.partition_key}" if self.partition_key else ""
+        return f"indexer{formatted_partition_key}:org:str:{cache_namespace}:{hashed}"
```

Comment:
I think it's a bit suboptimal that we have introduced two different formats of cache key now. What's the use case for not having partition key

Reply:
I think that's what was talking about in https://github.com/getsentry/sentry/pull/37632#discussion_r941847068 and why i did not have the additional `:` to begin with. I put in the description that I chose to make partition key optional if we didn't want to have to specify a db, or if we wanted to keep the original format